art/attacks/inference/attribute_inference/true_label_baseline.py

```diff
@@ -240,7 +229,105 @@ def fit(self, x: np.ndarray, y: np.ndarray) -> None:
+
+                        super().__init__()
+
+                        self.features = nn.Sequential(
+                            nn.Linear(self.num_features, 100),
+                            nn.ReLU(),
+                            nn.Linear(100, 64),
+                            nn.ReLU(),
+                            nn.Linear(64, 1),
+                        )
+
+                    def forward(self, x):
+                        """Forward the model."""
+                        return self.features(x)
+
+                self.attack_model = MembershipInferenceAttackModelRegression(x_train.shape[1])
+                loss_fn: Any = nn.MSELoss()
+            else:
+
+                class MembershipInferenceAttackModel(nn.Module):
+                    """
+                    Implementation of a pytorch model for learning an attribute inference attack.
+
+                    The features used are the remaining n-1 features of the attack training data along with
+                    the model's predictions.
+                    """
+
+                    def __init__(self, num_features, num_classes):
+
+                        self.num_classes = num_classes
+                        self.num_features = num_features
+
+                        super().__init__()
+
+                        self.features = nn.Sequential(
+                            nn.Linear(self.num_features, 512),
+                            nn.ReLU(),
+                            nn.Linear(512, 100),
+                            nn.ReLU(),
+                            nn.Linear(100, 64),
+                            nn.ReLU(),
+                            nn.Linear(64, num_classes),
+                        )
+
+                        self.output = nn.Softmax()
+
+                    def forward(self, x):
+                        """Forward the model."""
+                        out = self.features(x)
+                        return self.output(out)
+
+                self.attack_model = MembershipInferenceAttackModel(x_train.shape[1], len(self._values))
+                loss_fn = nn.CrossEntropyLoss()
+
+            optimizer = optim.Adam(self.attack_model.parameters(), lr=self.learning_rate)  # type: ignore
+
+            attack_train_set = self._get_attack_dataset(feature=x_train, label=y_ready)
+            train_loader = DataLoader(attack_train_set, batch_size=self.batch_size, shuffle=True, num_workers=0)
+
+            self.attack_model = to_cuda(self.attack_model)  # type: ignore
+            self.attack_model.train()  # type: ignore
+
+            for _ in range(self.epochs):
+                for (input1, targets) in train_loader:
+                    input1, targets = to_cuda(input1), to_cuda(targets)
+                    _, targets = torch.autograd.Variable(input1), torch.autograd.Variable(targets)
+
+                    optimizer.zero_grad()
+                    outputs = self.attack_model(input1)  # type: ignore
```

Comment:
Why is this type ignore needed?

Reply:
It was there before (in other classes), I just copied the code. Not sure why it's needed (I can tru to remove and see if it causes any errors).

Reply:
Removing it causes a mypy error: "Tensor" not callable

Reply:
Ok, leave it for now.